packages/react-devtools-shared/src/backend/views/Highlighter/Highlighter.js

```diff
@@ -41,13 +45,43 @@ export function showOverlay(
     return;
   }
 
-  if (overlay === null) {
-    overlay = new Overlay();
+  if (highlightCanvas === null) {
+    highlightCanvas = new Canvas();
```

Comment:
I think maybe there's still some confusion. This is still a separate canvas instance. The original goal of #21777 was not just to use `HTMLCanvasElement` for both, but to _share a single canvas_ rather than creating two 😄 Is there a reason that's not possible?

Reply:
To do this, we need to get check if there is an existing instance of Canvas and if it does, then not create a new Canvas here, at Highlighter.js. Is there a way to get the current Canvas object at Highlighter.js?. We can get the Canvas node in window itself, but can we get the object?

One thing that I tried is, create a `new Canvas` in Highlighter.js and in the constructor of the Canvas, return early with the existing Canvas object, so the Canvas does not actually get created. Like this,
```
constructor() {
    instances.push(this);
    if(instances.length > 1) {
      instances.pop();
      // return the first instance, if a new Canvas is created
      return instances[0]
    }
// rest of the constructor where we container div for the canvas
}
```
Is there a better way to do this?

Reply:
I don't have a mental picture of exactly how this should be implemented, but my goal for writing the original issue was to:
(1) Combine the ways we're drawing these highlights so we don't have two totally separate approaches.
(2) Share a single highlighting utility that draws (any types of highlight) to a single Canvas (starting with the two types we have).